src/analysis/asymptotic_equivalent.lean

```diff
@@ -0,0 +1,265 @@
+  split; rintros ⟨φ, hφ, h⟩; [use (φ + 1), use (φ - 1)]; split,
+  { conv in (𝓝 _) { rw ← zero_add (1 : β) },
+    exact hφ.add (tendsto_const_nhds) },
+  { convert h.add (eventually_eq.refl l v); ext; simp [add_mul] },
+  { conv in (𝓝 _) { rw ← sub_self (1 : β) },
+    exact hφ.sub (tendsto_const_nhds) },
+  { convert h.sub (eventually_eq.refl l v); ext; simp [sub_mul] }
+end
+
+lemma is_equivalent.exists_eq_mul (huv : u ~[l] v) :
+  ∃ (φ : α → β) (hφ : tendsto φ l (𝓝 1)), u =ᶠ[l] φ * v :=
+is_equivalent_iff_exists_eq_mul.mp huv
+
+lemma is_equivalent_of_tendsto_one (hz : ∀ᶠ x in l, v x = 0 → u x = 0)
+  (huv : tendsto (u/v) l (𝓝 1)) : u ~[l] v :=
+begin
+  rw is_equivalent_iff_exists_eq_mul,
+  refine ⟨u/v, huv, hz.mono $ λ x hz', (div_mul_cancel_of_imp hz').symm⟩,
+end
+
+lemma is_equivalent_of_tendsto_one' (hz : ∀ x, v x = 0 → u x = 0) (huv : tendsto (u/v) l (𝓝 1)) :
+  u ~[l] v :=
+is_equivalent_of_tendsto_one (eventually_of_forall hz) huv
+
+lemma is_equivalent_iff_tendsto_one (hz : ∀ᶠ x in l, v x ≠ 0) :
+  u ~[l] v ↔ tendsto (u/v) l (𝓝 1) :=
+begin
+  split,
+  { intro hequiv,
+    have := hequiv.is_o.tendsto_0,
+    simp only [pi.sub_apply, sub_div] at this,
+    have key : tendsto (λ x, v x / v x) l (𝓝 1),
+    { exact (tendsto_congr' $ hz.mono $ λ x hnz, @div_self _ _ (v x) hnz).mpr tendsto_const_nhds },
+    convert this.add key,
+    { ext, simp },
+    { norm_num } },
+  { exact is_equivalent_of_tendsto_one (hz.mono $ λ x hnvz hz, (hnvz hz).elim) }
+end
+
+lemma is_equivalent.mul (htu : t ~[l] u) (hvw : v ~[l] w) : t * v ~[l] u * w :=
+begin
+  rw is_equivalent_iff_exists_eq_mul at *,
+  rcases htu with ⟨φ₁, hφ₁, h₁⟩,
+  rcases hvw with ⟨φ₂, hφ₂, h₂⟩,
+  rw ← one_mul (1 : β),
+  refine ⟨φ₁ * φ₂, hφ₁.mul hφ₂, _⟩,
+  convert h₁.mul h₂ using 1,
+  ext,
+  simp only [pi.mul_apply],
+  ac_refl
+end
+
+lemma is_equivalent.inv (huv : u ~[l] v) : (λ x, (u x)⁻¹) ~[l] (λ x, (v x)⁻¹) :=
+begin
+  rw is_equivalent_iff_exists_eq_mul at *,
+  rcases huv with ⟨φ, hφ, h⟩,
+  rw ← inv_one,
+  refine ⟨λ x, (φ x)⁻¹, tendsto.inv' hφ (by norm_num) , _⟩,
+  convert h.inv,
+  ext,
+  simp [mul_inv']
+end
+
+lemma is_equivalent.div (htu : t ~[l] u) (hvw : v ~[l] w) :
+  (λ x, t x / v x) ~[l] (λ x, u x / w x) :=
+htu.mul hvw.inv
+
+end normed_field
+
+section normed_linear_ordered_field
+
+variables {α β : Type*} [normed_linear_ordered_field β] {u v : α → β} {l : filter α}
+
+lemma is_equivalent.tendsto_at_top [order_topology β] (huv : u ~[l] v) (hu : tendsto u l at_top) :
+  tendsto v l at_top :=
+let ⟨φ, hφ, h⟩ := huv.symm.exists_eq_mul in
+tendsto.congr' h.symm ((mul_comm u φ) ▸ (tendsto_mul_at_top zero_lt_one hu hφ))
+
+lemma is_equivalent.tendsto_at_top_iff [order_topology β] (huv : u ~[l] v) :
+  tendsto u l at_top ↔ tendsto v l at_top := ⟨huv.tendsto_at_top, huv.symm.tendsto_at_top⟩
+
+end normed_linear_ordered_field
+
+section smul
+
+lemma is_equivalent.smul {α E 𝕜 : Type*} [nondiscrete_normed_field 𝕜] [normed_group E]
```

Comment:
```suggestion
lemma is_equivalent.smul {α E 𝕜 : Type*} [normed_field 𝕜] [normed_group E]
```
I can't see where the nondiscreteness would matter. Also, can you move this earlier in the file, to deduce `is_equivalent.mul` from this one?

Reply:
Yes that works, thanks !